src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt

```diff
@@ -0,0 +1,31 @@
+package org.consoletrader.candles.bitmarket
+
+import io.reactivex.Observable
+import org.consoletrader.candles.Candle
+import org.consoletrader.candles.CandlesService
+import org.consoletrader.candles.base.BaseApi
+import org.knowm.xchange.currency.CurrencyPair
+import java.time.Instant
+
+class BitmarketCandleService : BaseApi<BitmarketPublicApi>(
+        anApi = BitmarketPublicApi::class.java,
+        endpoint = "https://www.bitmarket.pl/"),
+        CandlesService {
+
+    override fun getCandles(pair: CurrencyPair): Observable<Candle> {
+
+        //todo: pair is correct???
+        return getApi().queryCandles(pair = pair.toString())
+                .map {
```

Comment:
CurrencyPair.toString() will return BTC/PLN but bit market is expecting BTCPLN (no slash). You probably need to glue this "${pair.base}${pair.counter}"